pkg/queue/manager_test.go

```diff
@@ -160,6 +160,63 @@ func TestUpdateQueue(t *testing.T) {
 	}
 }
 
+func TestDeleteFromQueue(t *testing.T) {
+	now := time.Now()
+	cq := utiltesting.MakeClusterQueue("cq").Obj()
+	q := utiltesting.MakeQueue("foo", "").ClusterQueue(cq.Name).Obj()
+	wl1 := utiltesting.MakeWorkload("wl1", "").Queue(q.Name).Creation(now.Add(time.Hour)).Obj()
+	wl2 := utiltesting.MakeWorkload("wl2", "").Queue(q.Name).Creation(now.Add(time.Hour)).Obj()
+	wl3 := utiltesting.MakeWorkload("wl3", "").Queue(q.Name).Creation(now).Obj()
+	wl4 := utiltesting.MakeWorkload("wl4", "").Queue(q.Name).Creation(now).Obj()
+	workloads := []*kueue.Workload{wl1, wl2}
+	inadmissibleWorkloads := []*kueue.Workload{wl3, wl4}
+
+	scheme := runtime.NewScheme()
+	if err := kueue.AddToScheme(scheme); err != nil {
+		t.Fatalf("Failed adding kueue scheme: %s", err)
+	}
+	ctx := context.Background()
+	manager := NewManager(fake.NewClientBuilder().WithScheme(scheme).WithObjects(cq, q, wl1, wl2, wl3, wl4).Build())
+
+	if err := manager.AddClusterQueue(ctx, cq); err != nil {
+		t.Fatalf("Failed adding clusterQueue %s: %v", cq.Name, err)
+	}
+	cqImpl, exists := manager.clusterQueues[cq.Name]
+	if !exists {
+		t.Errorf("clusterQueue %s not exist", cq.Name)
+	}
+
+	if err := manager.AddQueue(ctx, q); err != nil {
+		t.Fatalf("Failed adding queue %s: %v", q.Name, err)
+	}
+	qImpl, exists := manager.queues[Key(q)]
+	if !exists {
+		t.Errorf("queue %s not exist", q.Name)
+	}
+
+	for _, w := range workloads {
+		manager.AddOrUpdateWorkload(w)
+	}
+	// simulate scheduling inadmissibleWorkloads
+	for i := 0; i < len(inadmissibleWorkloads); i++ {
+		cqImpl.Pop()
+	}
+	for _, w := range inadmissibleWorkloads {
+		manager.RequeueWorkload(ctx, workload.NewInfo(w), false)
+	}
+	if pending := cqImpl.Pending(); pending != 4 {
+		t.Errorf("clusterQueue's workload number not right, want %v, got %v", 4, pending)
+	}
+	if gotWorkloads, _ := cqImpl.Dump(); len(gotWorkloads) != 2 {
```

Comment:
I would remove this check. Dump should probably return everything, so I wouldn't test here that it is equal to only the admissible workloads.

Reply:
This is another question I have in mind, now `Pending` returns all the workloads both in heap and `inadmissibleWorkloads`, should `Dump` also returns everythin?

Reply:
I'll follow this later.